hi-ml-cpath/testhisto/testhisto/utils/test_tiles_selection_utils.py

```diff
@@ -167,13 +160,21 @@ def test_aggregate_shallow_slide_nodes(n_classes: int, rank: int = 0, world_size
 
     if rank == 0:
         for label in range(n_classes):
+
+            assert all(slide_node.pred_label == slide_node.true_label for slide_node in shallow_top_slides_heaps[label])
+            selected_top_slides_ids = [slide_node.slide_id for slide_node in shallow_top_slides_heaps[label]]
             expected_top_slides_ids = _get_expected_slides_by_probability(results, num_top_slides, label, top=True)
-            assert expected_top_slides_ids == [slide_node.slide_id for slide_node in shallow_top_slides_heaps[label]]
+            assert expected_top_slides_ids == selected_top_slides_ids
 
+            assert all(
+                slide_node.pred_label != slide_node.true_label for slide_node in shallow_bottom_slides_heaps[label]
+            )
+            selected_bottom_slides_ids = [slide_node.slide_id for slide_node in shallow_bottom_slides_heaps[label]]
             expected_bottom_slides_ids = _get_expected_slides_by_probability(results, num_top_slides, label, top=False)
-            assert expected_bottom_slides_ids == [
-                slide_node.slide_id for slide_node in shallow_bottom_slides_heaps[label]
-            ]
+            assert expected_bottom_slides_ids == selected_bottom_slides_ids
+
+            # Make sure that the top and bottom slides are disjoint.
+            assert not set(selected_top_slides_ids).intersection(set(selected_bottom_slides_ids))
```

Comment:
I don't think the current test reproduces the error you fixed with this PR, because the mock data distribution is as before. Do you think we could add an edge case test that emulates the situation small validation set (with respect to the number of slides asked) and poor performance to avoid getting again similar bugs in the future? for example an idea could be to:

-  parametrize the tests to have num_top_slides=90% of tot_slides, 
- parametrize the test to have an edge case where the predicted probability is always around 0.5

Also are there cases where we should raise errors? e.g. number of required top tiles >= total number of slides. Can we add assertions for these cases?

Reply:
Very good point, we can parametrize the num_top_slides as you suggested. 

About the edge case: the tests already cover it implicitly we sample 20 random slides some of which have probabilities around 0.5 and lower with n_classes>2. I had to fix one edge case where the probabilities were [0.3, 0.3, 0.4] and 0.4 were among lower probabibilities in the batch but corresponded to the predicted label so it shouldn't be selected in bottom slides.
this is why I added the `_selection_condition` to the expected slides to make sure slides with highest/lowest probs correspond to predicted labels.

Reply:
I see you parametrized the number of slides - good enough, let's get this merged